onset_maproom/tests/test_calc.py

```diff
@@ -4,24 +4,2025 @@
+        0.130293,
+        0.0,
+        0.20588,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.154896,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.255222,
+        0.126956,
+        0.10043,
+        0.006866,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.069079,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.001652,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.211269,
+        5.749138,
+        5.437428,
+        3.361074,
+        0.001031,
+        0.085564,
+        0.078808,
+        2.277596,
+        0.114164,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        4.02564,
+        1.322553,
+        0.0,
+        0.2692,
+        0.0,
+        0.0,
+        1.737253,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.451228,
+        0.054314,
+        0.333831,
+        0.637705,
+        6.406705,
+    ]
+    precip = xr.DataArray(values, dims=["T"], coords={"T": t}).rename("precip")
+
+    dts = calc.daily_tobegroupedby_season(precip, 29, 11, 29, 2)
+
+    assert dts["T"].size == 461
+    assert dts["group"].size == 5
+    assert (
+        dts.seasons_ends
+        == xr.DataArray(
+            [
+                "2001-02-28T00:00:00.000000000",
+                "2002-02-28T00:00:00.000000000",
+                "2003-02-28T00:00:00.000000000",
+                "2004-02-29T00:00:00.000000000",
+                "2005-02-28T00:00:00.000000000",
+            ]
+        )
+    ).all
+
+    onsetsds = calc.seasonal_onset_date(
+        precip, 1, 3, 90, 1, 3, 20, 1, 7, 21, time_coord="T"
+    )
+    onsets = onsetsds.onset_delta + onsetsds["T"]
+    assert (
+        onsets
+        == xr.DataArray(
+            [
+                "NaT",
+                "2001-03-08T00:00:00.000000000",
+                "NaT",
+                "2003-04-12T00:00:00.000000000",
+                "2004-04-04T00:00:00.000000000",
+            ]
+        )
+    ).all
```

Comment:
Did you actually verify that these results are correct? Even if you did, it's hard for the reader (me) to verify it for himself. I think it would be easier if you used synthetic data.

I'm not saying this test is worthless. It works as a regression test: if one day we change the function in a way that changes its outputs when we didn't intend to, this test can alert us to it. But it doesn't help (as much as it could) to convince us that the function is correct in the first place.

BTW it might be good to have one test that uses (a small amount of) actual data, in addition to ones that use synthetic data, to guard against writing code that is wrong but passes all the tests, because the code and the tests were both based on the same misunderstanding of what the data are supposed to look like.